### PR TITLE
DOCS-330: Continuing Clean Up Work

### PR DESCRIPTION
**Pull request by ravindk89**
# Summary

This is the last component of DOCS-330. This should include all core information previously covered between github and docs.min.io. 

# Goals

- Unified active-passive and active-active security sections, minor touch-ups
- Imported sections on behavior for server-side replication, delete replication, sync/async, etc.

# Non-Goals

- More diagrams
- Examples for `mc replicate`

# Build Instructions

The MinIO docs require Python 3.9+ and Node for building. For Node, use v14.5.0 or later.

1) Clone the repo, Checkout the branch (`git checkout -B DOCS-324 origin/DOCS-324`)
2) Run `npm install` from the branch
3) Create a python venv (`python -m venv venv`)
4) Install requirements (`pip install -r requirements.txt`)
5) Run `make html`
6) Run `python -m http.server --directory build/DOCS-324/html`
7) Open 127.0.0.1:8000 in your browser and take a poke around.

If you run into an error running `python -m http.server` related to `--directory`, you're python version is likely older. CD into the `build/<branch>/html` directory and run the command without the `--directory` argument.

# Key URLS

- http://127.0.0.1:8000/replication/replication-overview.html
- http://127.0.0.1:8000/replication/enable-server-side-two-way-bucket-replication.html
- - http://127.0.0.1:8000/replication/enable-server-side-one-way-bucket-replication.html

### CODE REVIEW

**Comment by ravindk89** [comment]
@poornas @vadmeste there is one more "chunk" missing here related to the replication queue. Copying over the gaps from slack:

1. Is the replication queue global or per bucket?
2. Is the replication queue size configurable?
3. Is the 10K queue size for all 100 workers, or per worker? 
4. What are the potential ramifications of increasing workers? 
5. Does synchronous replication rely on the replication queue? That is, if the queue is full, will synchronized replication wait until a worker gets to it? 
6. Is the replication queue FIFO ? that is, workers move in order of the queue from first to last.

**Review by poornas — CHANGES_REQUESTED** [review]

File: source/replication/enable-server-side-one-way-bucket-replication.rst [hunk 1]
```diff
@@ -166,23 +195,38 @@ source and destination clusters:
                         "s3:GetObjectVersion",
                         "s3:GetObjectVersionTagging",
                         "s3:PutObject",
+                        "s3:PutObjectRetention",
+                        "s3:PutObjectLegalHold",
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/replication/enable-server-side-one-way-bucket-replication.rst:199]: Done in both procedures

File: source/replication/enable-server-side-two-way-bucket-replication.rst [hunk 2]
```diff
@@ -14,18 +14,27 @@ Enable Two-Way Server-Side Bucket Replication
...
 
 - To configure one-way "active-passive" replication between MinIO clusters,
   see :ref:`minio-bucket-replication-serverside-oneway`.
 
+MinIO Active-Active replication is designed for two MinIO clusters. Multi-site
+replication adds significant complexity and should be undertaken in
+conjunction with the MinIO team. Enterprises looking to implement multi-site
+replication should consider leveraging `MinIO SUBNET
+<https://min.io/pricing?ref=docs>`__ support to access the expertise and
+planning required for this implementation.
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/replication/enable-server-side-two-way-bucket-replication.rst:36]: Refined based on convo in slack.

**Review by poornas — APPROVED** [review]
LGTM

**Pull request merged**